### ISSUE TITLE
Add warmup stage before starting the measurements

### ISSUE THREAD

**Issue by Thomas-Yang** [brianfrankcooper/YCSB#422]
I think it would be more rigorous to add a configurable warmup stage before starting the actual latency/throughput measurements, which
*  avoids the cold start effect at the backend
*  allows to customize how long (or what kinds of) the warmup should run

I found this feature was partially implemented in [(source)](https://github.com/jbellis/YCSB/blob/master/core/src/main/java/com/yahoo/ycsb/Client.java#L201), which could be a good starting point. 
Any thoughts?

**Comment by busbey** [comment]
yeah, this is sitting over in PR #121. The unrelated formatting changes made it hard to review. AFAICT, the original contributor was no longer interested due to how long it sat idle.

if you're interested in building on the commit in that earlier PR I think it'd be a great addition.
